lib/improver/utilities/cube_checker.py

```diff
@@ -144,48 +143,42 @@ def check_cube_coordinates(cube, new_cube, exception_coordinates=None):
     for coord in new_cube.aux_coords[::-1]:
         if coord.name() in cube_dim_names:
             new_cube = iris.util.new_axis(new_cube, coord)
+    new_cube_dim_names = [coord.name() for coord in new_cube.dim_coords]
+    # If we have the wrong number of dimensions then raise an error.
+    if (len(cube.dim_coords)+len(exception_coordinates) !=
+            len(new_cube.dim_coords)):
 
-    # Ensure dimension order matches; if lengths are unequal a coordinate
-    # is missing, so raise an appropriate error.
-    cube_dimension_order = {coord.name(): cube.coord_dims(coord.name())[0]
-                            for coord in cube.dim_coords}
+        msg = ('The number of dimension coordinates within the new cube '
+               'do not match the number of dimension coordinates within the '
+               'original cube plus the number of exception coordinates. '
+               '\n input cube dimensions {}, new cube dimensions {}'.format(
+                   cube_dim_names, new_cube_dim_names))
+        raise CoordinateNotFoundError(msg)
 
+    # Ensure dimension order matches
+    new_cube_dimension_order = {coord.name(): new_cube.coord_dims(
+        coord.name())[0] for coord in new_cube.dim_coords}
     correct_order = []
     new_cube_only_dims = []
-    new_cube_dim_names = [coord.name() for coord in new_cube.dim_coords]
-    for coord_name in new_cube_dim_names:
-        if coord_name in cube_dim_names:
-            correct_order.append(cube_dimension_order[coord_name])
-        else:
-            if coord_name in exception_coordinates:
-                new_coord_dim = new_cube.coord_dims(coord_name)[0]
-                new_cube_only_dims.append(new_coord_dim)
-            else:
-                msg = ("The coordinate: {0} is within new_cube, "
-                       "however, this is not within the original "
-                       "cube. As {0} is not within the permitted "
-                       "exceptions: {1}, this is not allowed. "
-                       "\nnew_cube: {2}"
-                       "\ncube: {3}").format(
-                           coord_name, exception_coordinates, new_cube,
-                           cube)
-                raise InvalidCubeError(msg)
+    for coord_name in cube_dim_names:
+        correct_order.append(new_cube_dimension_order[coord_name])
+    for coord_name in exception_coordinates:
+        try:
+            new_coord_dim = new_cube.coord_dims(coord_name)[0]
+            new_cube_only_dims.append(new_coord_dim)
+        except CoordinateNotFoundError:
+            msg = ("The coordinate: {0} is not new_cube, "
```

Comment:
Maybe this could be `All permitted exceptions must be on the new_cube. In this case, coordinate {0} within the list of permitted exceptions ({1}) is not available on the new_cube.`?

Reply:
Changed